staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1031,17 +1031,34 @@ func (e *Store) DeleteCollection(ctx context.Context, deleteValidation rest.Vali
 				errs <- fmt.Errorf("DeleteCollection goroutine panicked: %v", panicReason)
 			})
 			defer wg.Done()
+			errorsSeen := 0
+			errorsReported := 0
+			errorsThreshold := 9
 
 			for index := range toProcess {
 				accessor, err := meta.Accessor(items[index])
 				if err != nil {
-					errs <- err
-					return
+					errorsSeen++
```

Comment:
this is the wrong spot to try to aggregate errors. Errors are getting aggregated below after the wait group completes, that structure makes multi-error reporting per go-func risky for deadlock.  That's the area that needs fixing. Not this loop.

Reply:
You're suggesting to do error aggregation around line 1067:
```
	select {
	case err := <-errs:
```
I would do this in next push where error count and last error is returned.